packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -298,12 +353,38 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
     _extent = _DraggableSheetExtent(
       minExtent: widget.minChildSize,
       maxExtent: widget.maxChildSize,
+      snap: widget.snap,
+      snapSizes: _impliedSnapSizes(),
       initialExtent: widget.initialChildSize,
       listener: _setExtent,
     );
     _scrollController = _DraggableScrollableSheetScrollController(extent: _extent);
   }
 
+  List<double> _impliedSnapSizes() {
+    for (int index = 0; index < (widget.snapSizes?.length ?? 0); index += 1) {
+      final double snapSize = widget.snapSizes![index];
+      assert(snapSize >= widget.minChildSize && snapSize <= widget.maxChildSize,
+        'Snap sizes must be between `minChildSize` and `maxChildSize`. Invalid size: $snapSize');
```

Comment:
nit: here and below, probably good to dump the whole list and point to the invalid one rather than just printing the invalid one, which might occur multiple times in a badly formed list.

Reply:
Fix'd!

```
Invalid snapSize '0.4' at index 3 of:
  [0.3, 0.4, 0.5, >>> 0.4 <<<, 0.6, 0.7]
Snap sizes must be in ascending order.
```

However, I'm getting the canned Flutter message "Either the assertion indicators an error in the framework itself, or we should..." message. It looks like it's just naively checking if the top two frames of the stack are Flutter and not user code? Do you know how to get rid of this?

Reply:
Feel free to file a bug for that - that code has a history of breaking :\

Reply:
File a bug for moving the asserts into the constructor, or file a bug for the "Either the assertion..." message firing?

Reply:
For the `Either the assertion` message :)